src/app/Rules/Rules.tsx

```diff
@@ -0,0 +1,205 @@
+        info({
+          tooltip: 'Period in seconds. Cryostat will connect to matching targets at this interval and copy the relevant recording data into its archives. Values less than 1 prevent data from being copied into archives - recordings will be started and remain only in target JVM memory.'
+        })
+      ],
+    },
+    {
+      title: 'Preserved Archives',
+      transforms: [
+        info({
+          tooltip: 'The number of recording copies to be maintained in the Cryostat archives. Cryostat will continue retrieving further archived copies and trimming the oldest copies from the archive to maintain this limit. Values less than 1 prevent data from being copied into archives - recordings will be started and remain only in target JVM memory.'
+        })
+      ],
+    },
+    {
+      title: 'Maximum Age',
+      transforms: [
+        info({
+          tooltip: 'The maximum age in seconds for data kept in the JFR recordings started by this rule. Values less than 1 indicate no limit.'
+        })
+      ],
+    },
+    {
+      title: 'Maximum Size',
+      transforms: [
+        info({
+          tooltip: 'The maximum size in bytes for JFR recordings started by this rule. Values less than 1 indicate no limit.'
+        })
+      ],
+    },
+  ] as ICell[];
+
+  const refreshRules = React.useCallback(() => {
+    setIsLoading(true);
+    addSubscription(
+      context.api.doGet('rules', 'v2').subscribe((v: any) => {
+        setRules(v.data.result);
+        setIsLoading(false);
+      })
+    );
+  }, [setIsLoading, addSubscription, context, context.api, setRules]);
+
+  React.useEffect(() => {
+    refreshRules();
+  }, [context, context.api]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel.messages(NotificationCategory.RuleCreated)
+        .subscribe(v => setRules(old => old.concat(v.message)))
+    );
+  }, [addSubscription, context, context.notificationChannel, setRules]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel.messages(NotificationCategory.RuleDeleted)
+        .subscribe(v => setRules(old => old.filter(o => o.name != v.message.name)))
+    )
+  }, [addSubscription, context, context.notificationChannel, setRules]);
+
+  React.useEffect(() => {
+    if (!context.settings.autoRefreshEnabled()) {
+      return;
+    }
+    const id = window.setInterval(() => refreshRules(), context.settings.autoRefreshPeriod() * context.settings.autoRefreshUnits());
+    return () => window.clearInterval(id);
+  }, []);
+
+  const handleSort = (event, index, direction) => {
+    setSortBy({ index, direction });
+  };
+
+  const displayRules = React.useMemo(() => {
+    const { index, direction } = sortBy;
+    let sorted = [...rules];
+    if (typeof index === 'number') {
+      const keys = ['name'];
+      const key = keys[index];
+      sorted = rules
+        .sort((a: Rule, b: Rule): number => (a[key] < b[key] ? -1 : a[key] > b[key] ? 1 : 0));
```

Comment:
Yea, it's a little bit unnecessary since there is only (currently) one sortable column here, but that's the gist of it. `a[key]` is equivalent to `a.name`. When the user clicks on the column header it emits an event that lets us track which column we should sort by, so that's what gives us our `index`. We map that to a property name of the `Rule` objects and sort on it.